shaarli_client/utils.py

```diff
@@ -21,3 +22,27 @@ def generate_all_endpoints_parsers(subparsers, endpoints):
     """Generate all endpoints' subparsers from an endpoints dict"""
     for ep_name, ep_metadata in endpoints.items():
         generate_endpoint_parser(subparsers, ep_name, ep_metadata)
+
+
+def write_output(args, response):
+    """Write the API response to file or standard output"""
+    if not args.outfile:
```

Comment:
printing to console and writing to files should be split over two separate functions, with a bit of logic to process argparse flags in the `main()` function